### PR TITLE
Add twin array tests using custom object for property values

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

**Comment by azabbasi** [comment]
I don't see a test being added am I missing something?

File: e2e/test/iothub/twin/TwinE2ETests.cs [hunk 1]
```diff
@@ -119,25 +121,25 @@ public async Task Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEvent_AmqpWs()
         [LoggedTestMethod]
         public async Task Twin_ServiceSetsDesiredPropertyArrayAndDeviceReceivesEvent_Mqtt()
         {
-            await Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEventAsync(Client.TransportType.Mqtt_Tcp_Only, SetTwinPropertyUpdateCallbackHandlerAsync, JArray.Parse("[1, \"someString\", false]")).ConfigureAwait(false);
+            await Twin_ServiceSetsDesiredPropertyAndDeviceReceivesEventAsync(Client.TransportType.Mqtt_Tcp_Only, SetTwinPropertyUpdateCallbackHandlerAsync, _listOfPropertyValues).ConfigureAwait(false);
```
Comment (drwill-ms) [inline e2e/test/iothub/twin/TwinE2ETests.cs:124]: These lines are getting long. Consider breaking them up?

File: e2e/test/iothub/twin/TwinE2ETests.cs [hunk 2]
```diff
@@ -581,4 +583,12 @@ await deviceClient
             Assert.IsFalse(serviceTwin.Properties.Reported.Contains(propName1));
         }
     }
+
+    internal class CustomTwinProperty
```
Comment (azabbasi) [inline e2e/test/iothub/twin/TwinE2ETests.cs:587]: I think you left out some parts of the change from this commit.

File: e2e/test/iothub/twin/TwinE2ETests.cs [hunk 3]
```diff
@@ -19,6 +20,7 @@ namespace Microsoft.Azure.Devices.E2ETests.Twins
     public class TwinE2ETests : E2EMsTestBase
     {
         private readonly string _devicePrefix = $"E2E_{nameof(TwinE2ETests)}_";
+        private readonly List<object> _listOfPropertyValues = new List<object> { 1, "someString", false, new CustomTwinProperty { Id = 123, Name = "someName" } };
```
Comment (drwill-ms) [inline e2e/test/iothub/twin/TwinE2ETests.cs:23]: Consider formatting this for easier reading:

```csharp
private readonly List<object> _listOfPropertyValues = List<object>
{
    1,
    "someString",
    //...
};
```

Comment (drwill-ms) [inline e2e/test/iothub/twin/TwinE2ETests.cs:23]: static?

Comment (abhipsaMisra) [reply e2e/test/iothub/twin/TwinE2ETests.cs:23]: sure

**Review by drwill-ms — APPROVED** [review]

File: e2e/test/iothub/twin/TwinE2ETests.cs (hunk 1 above)
Comment (abhipsaMisra) [reply e2e/test/iothub/twin/TwinE2ETests.cs:124]: sure

File: e2e/test/iothub/twin/TwinE2ETests.cs (hunk 2 above)
Comment (abhipsaMisra) [reply e2e/test/iothub/twin/TwinE2ETests.cs:587]: umm, what do you mean?

Comment (azabbasi) [reply e2e/test/iothub/twin/TwinE2ETests.cs:587]: ignore me. I totally missed out on the static property you added that uses this class.

**Review by azabbasi — APPROVED** [review]

**Pull request merged**